src/EnergyPlus/WeatherManager.cc

```diff
@@ -3480,8 +3480,8 @@ namespace WeatherManager {
 
         // Do the first five.  (To get to the DataSource field)
         {
-            auto nth_pos = nth_occurrence(current_line, ',', 5);
-            const bool succeeded = readList(current_line.substr(pos, nth_pos), WYear, WMonth, WDay, WHour, WMinute);
+            auto nth_pos = nth_occurrence(current_line, ',', 5); // Returns the position **after** the nth occurrence of ','
+            const bool succeeded = readList(current_line.substr(pos, (nth_pos - 1) - pos), WYear, WMonth, WDay, WHour, WMinute);
```

Comment:
When using nth_occurrence, you want to chop with -1

Reply:
Please don't use `auto` for scalars. Leave it for iterators and object references. I know you didn't put it in there, but don't perpetuate this practice.

Reply:
What do you suggest I replace `auto` with here?

* `std::string_view::size_type` to match the above declaration of the `pos` at the beginning of the routine? (Is that really clearer than `auto` then?)
* `std::size_t`?

----

`std::string_view::size_type` is the actual deduced return type for `nth_occurrence`: https://github.com/NREL/EnergyPlus/blob/fcbb8be256c8d5d408a763b899bd494d9a4ec231/src/EnergyPlus/StringUtilities.hh#L122-L158

FYI, ObjexxFCL's `index` returns a `std::string::size_type` (thought it takes a string_view as input and uses find on it). https://github.com/NREL/EnergyPlus/blob/98f434ed153c8162441d47566bf2e98d293343bd/third_party/ObjexxFCL/src/ObjexxFCL/string.functions.hh#L710-L715

* `std::string_view::size_type` is typedefed to `std::size_t`
* `std::string::size_type` is [std::allocator_traits](http://en.cppreference.com/w/cpp/memory/allocator_traits)<Allocator>::size_type, which eventually resolves to `std::size_t` as well

Reply:
Used std::string_view::size_type in 9e00e83c7175fd09447a3fe1e8f9ff423a90b29c